app/src/main/java/com/duckduckgo/app/global/view/ActivityExtension.kt

```diff
@@ -64,7 +71,20 @@ fun FragmentActivity.toggleFullScreen() {
     window.decorView.systemUiVisibility = newUiOptions
 }
 
+fun FragmentActivity.isFullScreen(): Boolean {
+    return window.decorView.systemUiVisibility.and(View.SYSTEM_UI_FLAG_FULLSCREEN) == View.SYSTEM_UI_FLAG_FULLSCREEN
+}
+
 fun FragmentActivity.isImmersiveModeEnabled(): Boolean {
     val uiOptions = window.decorView.systemUiVisibility
     return uiOptions or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY == uiOptions
 }
+
+fun FragmentActivity.requestJsOrientationChange(newOrientation: JavaScriptScreenOrientation): Boolean {
+    if (!isFullScreen()) {
+        return false
+    }
+
+    requestedOrientation = newOrientation.nativeValue
+    return true
+}
```

Comment:
There is no need to expose `JavaScriptScreenOrientation`. Pass an int instead to make it generic
```suggestion
fun FragmentActivity.requestJsOrientationChange(newOrientation: Int): Boolean {
    if (!isFullScreen()) {
        return false
    }

    requestedOrientation = newOrientation
    return true
}
```

Reply:
Won't that allow any int to be passed? Don't we want the values to be restricted to allowed orientation values? I could add some sanity checks, but that will add another possible error here...

Reply:
Your API is a bit convoluted/odd. 
* naming: what you're doing is mapping js string orientation to android screen (activity info) orientation, yet the enum is called `JavaScriptScreenOrientation`
* the extension function that requests the orientation is called `requestJsOrientationChange` yet it doesn't have anything to do with JS, its Android
* the extension function depends on a type (`JavaScriptScreenOrientation`) defined in a very different place

What I would do
* rename `requestJsOrientationChange` to `requestOrientationChangeFromJsOrientation`
* signature be the one below and perform the checks inside it
```kotlin
fun FragmentActivity.requestJsOrientationChange(newOrientation: String): Result<Boolean>
```
* Modify `updateOrientation` it just calls `requestJsOrientationChange(requestedOrientation)` and then handles the `Result<Boolean>` same way you handle now the errors

Reply:
True, it was a bit funky. I re-named and re-architected things a bit. Seems cleaner now, let me know what you think.